src/components/results/use-results-tab.ts

```diff
@@ -38,11 +38,85 @@ export type ResultTabIndexRedirection =
  */
 export const useResultsTab = (
     resultTabIndexRedirection: ResultTabIndexRedirection,
-    setTabIndex: (index: number) => void,
-    tabLevel: ResultsTabsLevel
-) => {
+    setTabIndex: React.Dispatch<React.SetStateAction<number>>,
+    view: string
+): Dispatch<SetStateAction<Boolean>> => {
+    const [redirectionLock, setRedirectionLock] =
+        useResultsTabRedirectionLock();
+
+    useEffect(() => {
+        if (view !== StudyView.RESULTS && !redirectionLock) {
+            setTabIndex(resultTabIndexRedirection);
+        }
+    }, [resultTabIndexRedirection, setTabIndex, view, redirectionLock]);
+
+    return setRedirectionLock;
+};
+
+const useResultsTabRedirectionLock = (): [
+    Boolean,
+    Dispatch<SetStateAction<Boolean>>
+] => {
+    const computationStatus: ComputingStatus = useSelector(
+        (state: ReduxState) => state.computingStatus
+    );
+    const [redirectionLock, setRedirectionLock] = useState<Boolean>(false);
```

Comment:
add some comments around "setRedirectionLock" usages to explain why we should lock/unlock redirections

Reply:
Done